llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -663,6 +682,8 @@ void SearchableTableEmitter::run(raw_ostream &OS) {
   // Emit tables in a deterministic order to avoid needless rebuilds.
   SmallVector<std::unique_ptr<GenericTable>, 4> Tables;
   DenseMap<Record *, GenericTable *> TableMap;
+  if (Records.getClass("Instruction"))
```

Comment:
```suggestion
  if (!Records.getAllDerivedDefinitions("Instruction").empty())
```
Do this only if there are `Instruction` definitions instead of `Instruction` class definition.

Reply:
Done.
I've added the derived definitions guard as suggested. I needed to leave in the original guard since the derived definitions lookup will fail with "The class 'Instruction' is not defined" for clients that do not happen to have a Instruction class at all.

If I could also trouble you (or anyone else) to merge on my behalf, I'd appreciate that.

Reply:
Oh sorry, I forgot it, we can use `getAllDerivedDefinitionsIfDefined` here.
> If I could also trouble you (or anyone else) to merge on my behalf, I'd appreciate that.

Yeah, no problem!

Reply:
Done (used `getAllDerivedDefinitionsIfDefined`).